data_override/include/data_override.inc

```diff
@@ -695,8 +784,71 @@ subroutine DATA_OVERRIDE_0D_(gridname,fieldname_code,data_out,time,override,data
+      dims = get_external_field_size(id_time)
+      prev_dims = get_external_field_size(id_time_prev)
+      ! check consistency of spatial dims
+      if ((prev_dims(1) .ne. dims(1)) .or. (prev_dims(2) .ne. dims(2)) .or. &
+          (prev_dims(3) .ne. dims(3))) then
+        call mpp_error(FATAL, 'data_override: dimensions mismatch between consecutive forcing files')
+      endif
+      allocate(data_table(index1)%time_prev_records(prev_dims(4)))
+      call get_time_axis(id_time_prev,data_table(index1)%time_prev_records)
+    endif if_prev1
+    id_time_next = -1
+    if_next1: if (trim(nextfilename) /= '') then
+      id_time_next = init_external_field(nextfilename,fieldname,verbose=.false.)
+      dims = get_external_field_size(id_time)
+      next_dims = get_external_field_size(id_time_next)
+      ! check consistency of spatial dims
+      if ((next_dims(1) .ne. dims(1)) .or. (next_dims(2) .ne. dims(2)) .or. &
+          (next_dims(3) .ne. dims(3))) then
+        call mpp_error(FATAL, 'data_override: dimensions mismatch between consecutive forcing files')
+      endif
+      allocate(data_table(index1)%time_next_records(next_dims(4)))
+      call get_time_axis(id_time_next,data_table(index1)%time_next_records)
+    endif if_next1
+  endif if_multi1
+
+
   !10 do time interp to get data in compute_domain
-  call time_interp_external(id_time, time, data_out, verbose=.false.)
+
+  first_record = data_table(index1)%time_records(1)
+  last_record = data_table(index1)%time_records(dims(4))
+
+  ! if using consecutive files, allow to perform time interpolation between the last record of previous
+  ! file and first record of current file OR between the last record of current file and first record of
+  ! next file hence "bridging" over files.
+  if_multi2: if (multifile) then
+    if_time2: if (time<first_record) then
+      if (id_time_prev<0) call mpp_error(FATAL,'data_override:previous file needed with multifile')
+      prev_dims = get_external_field_size(id_time_prev)
+      if (time<data_table(index1)%time_prev_records(prev_dims(4))) call mpp_error(FATAL, &
+          'data_override: time_interp_external_bridge should only be called to bridge with previous file')
+      call time_interp_external_bridge(id_time_prev, id_time,time,data_out,verbose=.false.)
+    elseif (time>last_record) then
+      if (id_time_next<0) call mpp_error(FATAL,'data_override:next file needed with multifile')
+      if (time>data_table(index1)%time_next_records(1)) call mpp_error(FATAL, &
+          'data_override: time_interp_external_bridge should only be called to bridge with next file')
+      call time_interp_external_bridge(id_time, id_time_next,time,data_out,verbose=.false.)
+    else ! first_record < time < last_record, do not use bridge
+      call time_interp_external(id_time,time,data_out,verbose=.false.)
+    endif if_time2
+  else ! standard behavior
+     if ((time<first_record) .or. (time>last_record)) then
+       call mpp_error(WARNING, &
```

Comment:
This WARNING is swamping the stdout for regression tests that were/are working fine. There is something wrong with this logic.